code/modules/wiremod/components/string/concat.dm

```diff
@@ -28,6 +34,22 @@
 
 /obj/item/circuit_component/concat/input_received(datum/port/input/port)
 	. = ..()
+
+	if(input_port_amount != current_option)
+		
+		if(input_port_amount > current_option)
+			for(var/i in current_option + 1 to input_port_amount)
+				var/datum/port/input/o = input_ports[current_option + 1]
+				remove_input_port(o)
+		
+		if(input_port_amount < current_option)
+			input_ports -= trigger_input
+			for(var/i in input_port_amount + 1 to current_option)
+				add_input_port(ascii2text(text2ascii("A") + (i-1)), PORT_TYPE_STRING)
```

Comment:
\+ 1 - 1

Reply:
Eliminating the +1-1 would make the code harder to read. We're adding input ports number input_port_amount+1 to current_option, not input ports input_port_amount to current_option-1. If we were using for(var/i=input_port_amount;i<current_option;i++) it could look alright but as it stands, it'd just look semantically worse.